src/components/structures/TimelinePanel.js

```diff
@@ -1023,7 +1020,16 @@ var TimelinePanel = React.createClass({
             var boundingRect = node.getBoundingClientRect();
             if ((allowPartial && boundingRect.top < wrapperRect.bottom) ||
                 (!allowPartial && boundingRect.bottom < wrapperRect.bottom)) {
-                return i;
+                // find last invisible event before the next visible one.
```

Comment:
It does, this extra thing is to then backtrack back down the timeline so that the RM always moves down the timeline. Say device B, a mobile app, does not have support for hidden events (yet), it could set the RM to latest event, then if on device A, riot-web with this support the RM could move up the timeline if the bottom event is hidden.